enterprise/web.jsf.editor/src/org/netbeans/modules/web/jsf/editor/completion/JsfAttributesCompletionHelper.java

```diff
@@ -461,7 +462,7 @@ public static void completeFaceletsFromProject(CompletionContext context, List<C
     public static void completeXMLNSAttribute(CompletionContext context, List<CompletionItem> items, JsfSupportImpl jsfs) {
         if (context.getAttributeName().toLowerCase(Locale.ENGLISH).startsWith("xmlns")) { //NOI18N
             //xml namespace completion for facelets namespaces
-            Set<String> nss = NamespaceUtils.getAvailableNss(jsfs.getLibraries(), jsfs.isJsf22Plus());
+            Set<String> nss = NamespaceUtils.getAvailableNss(jsfs.getLibraries(), jsfs.getJsfVersion().isAtLeast(JsfVersion.JSF_2_2));
```

Comment:
I think `getAvailableNss` needs to be changed to take the version. The boolean is used to determine if only the current namespace or also legacy ones should be reported. With JSF4 this would add a third set:

- \>= JSF4 would get the new namespace + legacy jcp + legacy sun
- \>= JSF2.2 && < JSF4 would get legacy jcp + legacy sun
- < JSF2.2 would get legacy sun

At least that is my interpretation from looking at it.

Reply:
I guess you're right.

Just for more context: Actually I was focusing on getting the functionality working mentioned in the isssue as I know what I can test. I did not do a all over the place, ensure that jsf 4 is supported refactoring.

I think this should be adjusted as well, but I'm not aware about the consequences and how to validate that behavior.

Reply:
This method is invoced when you have this situation:

```xml
<f:view
	xmlns="http://www.w3.org/1999/xhtml"
	xmlns:c="http:|"
</f:view>
```

The caret/cursor is placed at the location marked with "|" and you call for completion (I have mapped that to CTRL+Space. You should get a list of supported libraries/namespaces (screenshot from a project based on JSF 2.2):

![image](https://github.com/apache/netbeans/assets/2179736/f1eb4104-0826-4487-ac80-774b41ba0245)

You can see, that both the legacy namespaces and the "jcp" namespaces are returned. The order should be reversed from my POV, but that is a separate discussion. In a JSF4 project I would expect all namespaces as already described.